src/OauthTokenFileStorage.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge;
+
+use Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface;
+use Drupal\apigee_edge\Form\AuthenticationForm;
```

Comment:
This is why I questions whether that config name constant should be a public inside this class. Why a service depends on a form class? :) I haven't seen like this in Drupal core, also Drupal core does not define constant for config objects, not even in config forms. So this looks weird to me, but this can be fixed later.

Reply:
LGTM